src/components/Modal.js

```diff
@@ -60,8 +61,8 @@ const Modal = React.createClass({
 
   componentDidMount () {
   /*eslint-disable */
-    if (this.props.isOpen !== null) {
-      console.warn('WARNING: The prop "isOpen" is depracated in this version of the component. Please handle Modal opening from its parent.');
+    if (_has(this.props, 'isOpen')) {
```

Comment:
I had the same thought as @tumentumurchudur, and yes, it doesn't trigger. 

As for the hasOwnProperty, I read that is isn't completely reliable on IE. Not sure we are that concerned about that, though. Thoughts?